### ISSUE TITLE
Generate root-level serializer for statically known external serializers

### ISSUE THREAD

**Issue by qwwdfsad** [Kotlin/kotlinx.serialization#277]
```
@Serializer(forClass = Foo::class)
object FooSerializer : ... 

@Serializable(FooSerializer::class)
data class Foo()

Json.stringify(Foo())
```

will throw
```
kotlinx.serialization.SerializationException: Can't locate argument-less serializer for class kotlinx.serialization.json.JsonElement. For generic classes, such as lists, please provide serializer explicitly.
	at kotlinx.serialization.PlatformUtilsKt.serializer(PlatformUtils.kt:28)
	at kotlinx.serialization.context.ContextKt.getOrDefault(Context.kt:39)
```

Plugin should generate or augment `Foo` companion with `FooSerializer` as serializer.
It will significantly improve end user experience because "obvious" code will work without additional modules installations or explicit serializer arguments

**Comment by pdvrieze** [comment]
I would propose that is is only true if there is exactly one serializer or if class Foo actually announces its serializer. As an alternative it makes sense to generate an extension function `val Serializable.Companion.serializer: KSerializer<Serializable>` for those cases where the companion is not accessible. Access to the extension function would follow the normal name resolution (and conflict handling) rules.

**Comment by sandwwraith** [comment]
Implemented in Kotlin 1.3.30
